File: app/src/main/java/com/sumian/sd/account/userProfile/ModifyUserInfoContract.java
package com.sumian.sd.account.userProfile;


import androidx.annotation.NonNull;

import com.sumian.common.widget.picker.NumberPickerView;


/**
 * Created by sm
 * <p>
 * on 2018/7/5
 * <p>
 * desc:
 */
public interface ModifyUserInfoContract extends ImproveUserProfileContract {

    interface View extends ImproveUserProfileContract.View {

        void transformOneDisplayedValues(int currentPosition, String hintText, String[] displayedValues);

        void transformTwoDisplayedValues(int currentPosition, String hintText, String[] displayedValues);

        void transformThreeDisplayedValues(int currentPosition, String hintText, String[] displayedValues);

        void showOnePicker(int visible);

        void showTwoPicker(int visible);

        void showThreePicker(int visible);

        void transformTitle(@NonNull String title);

    }

    interface Presenter extends ImproveUserProfileContract.Presenter {

        void transformTitle(String modifyKey);

        String transformModify(@NonNull String modifyKey, @NonNull NumberPickerView pickerView, @NonNull NumberPickerView pickerTwo, @NonNull NumberPickerView pickerThree);

        void transformCityForProvince(@NonNull String province);

        void transformAreaForCity(@NonNull String city);
    }
}

File: patient/src/main/java/com/sumian/sd/buz/account/userProfile/ModifyUserInfoContract.kt
package com.sumian.sd.buz.account.userProfile


import com.sumian.common.widget.picker.NumberPickerView


/**
 * Created by sm
 *
 *
 * on 2018/7/5
 *
 *
 * desc:
 */
interface ModifyUserInfoContract : ImproveUserProfileContract {

    interface View : ImproveUserProfileContract.View {

        fun transformOneDisplayedValues(currentPosition: Int, hintText: String?, displayedValues: Array<out String?>?)

        fun transformTwoDisplayedValues(currentPosition: Int, hintText: String?, displayedValues: Array<out String?>?)

        fun transformThreeDisplayedValues(currentPosition: Int, hintText: String?, displayedValues: Array<out String?>?)

        fun showOnePicker(visible: Int)

        fun showTwoPicker(visible: Int)

        fun showThreePicker(visible: Int)

        fun transformTitle(title: String)

    }

    interface Presenter : ImproveUserProfileContract.Presenter {

        fun transformTitle(modifyKey: String)

        fun transformModify(modifyKey: String, pickerView: NumberPickerView, pickerTwo: NumberPickerView, pickerThree: NumberPickerView): String

        fun transformCityForProvince(province: String)

        fun transformAreaForCity(city: String)
    }
}
